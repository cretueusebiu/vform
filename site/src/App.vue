<template>
  <div class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
    <div class="max-w-5xl mx-auto">
      <div class="mt-4 flex items-start">
        <div>
          <div class="text-2xl font-normal text-indigo-600">
            <a href="/">vform</a>
          </div>
          <div class="mt-1 dark:text-gray-300">
            Handle Laravel-Vue forms and validation with ease.
          </div>
        </div>

        <ToggleTheme class="ml-auto my-auto" />
      </div>

      <Badges class="mt-2" />

      <div class="grid grid-cols-4 gap-5 mt-10">
        <div class="col-span-1">
          <Navigation />
        </div>

        <div class="col-span-3">
          <Demo class="mb-4" />

          <div class="panel">
            <Docs />
          </div>
        </div>
      </div>

      <div class="text-center py-4 text-gray-400">
        Created by <a href="https://github.com/cretueusebiu" target="_blank">Cretu Eusebiu</a> âœŒ
      </div>
    </div>
  </div>
  </div>
</template>

<script lang="ts">
import Docs from './../../docs.md'
import Demo from './components/Demo.vue'
import Badges from './components/Badges.vue'
import Navigation from './components/Navigation.vue'
import ToggleTheme from './components/ToggleTheme.vue'

export default {
  name: 'App',

  components: {
    Docs,
    Demo,
    Badges,
    Navigation,
    ToggleTheme
  }
}
</script>
