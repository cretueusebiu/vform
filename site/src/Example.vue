<template>
  <div>
    this is an Examples
    <button @click="click">
      click
    </button>
  </div>
</template>

<script>
export default {
  methods: {
    click () {
      alert('clicked')
    }
  }
}
</script>
