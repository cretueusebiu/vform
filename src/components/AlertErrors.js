import Alert from './Alert'

export default {
  name: 'AlertErrors',

  extends: Alert,

  props: {
    message: {
      type: String,
      default: 'There were some problems with your input.'
    }
  }
}
