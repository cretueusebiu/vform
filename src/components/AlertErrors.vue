<template>
  <div v-if="form.errors.any()" class="alert alert-danger alert-dismissible" role="alert">
    <button v-if="dismissible" type="button" class="close" aria-label="Close" @click="dismiss">
      <span aria-hidden="true">&times;</span>
    </button>

    <div v-if="message" v-html="message"/>

    <ul>
      <li v-for="error in form.errors.flatten()" v-html="error"/>
    </ul>
  </div>
</template>

<script>
import Alert from './Alert'

export default {
  name: 'alert-errors',

  extends: Alert,

  props: {
    message: {
      type: String,
      default: 'There were some problems with your input.'
    }
  }
}
</script>
