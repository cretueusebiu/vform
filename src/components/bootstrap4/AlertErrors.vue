<template>
  <div v-if="form.errors.any()" class="alert alert-danger alert-dismissible" role="alert">
    <button v-if="dismissible" type="button" class="close" aria-label="Dismiss" @click="dismiss">
      <span aria-hidden="true">&times;</span>
    </button>

    <div v-if="message" class="mb-2" v-html="message" />

    <ul class="mb-0">
      <li v-for="(error, index) in form.errors.flatten()" :key="index" v-html="error" />
    </ul>
  </div>
</template>

<script>
import Base from './../AlertErrors'

export default {
  extends: Base
}
</script>
