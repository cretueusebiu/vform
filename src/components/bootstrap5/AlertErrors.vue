<template>
  <div v-if="form.errors.any()" class="alert alert-danger alert-dismissible" role="alert">
    <button v-if="dismissible" type="button" class="btn-close" aria-label="Close" @click="dismiss" />

    <div v-if="message" class="mb-2" v-html="message" />

    <ul class="mb-0 ps-4">
      <li v-for="(error, index) in form.errors.flatten()" :key="index" v-html="error" />
    </ul>
  </div>
</template>

<script>
import Base from './../AlertErrors'

export default {
  extends: Base
}
</script>
