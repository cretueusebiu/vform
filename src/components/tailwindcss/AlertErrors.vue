<template>
  <div v-if="form.errors.any()" class="rounded-md bg-red-50 p-4 mb-3" role="alert">
    <div class="flex">
      <div>
        <slot>
          <div v-if="message" class="text-sm font-medium text-red-700" v-html="message" />

          <div class="mt-2 text-sm text-red-700">
            <ul class="list-disc pl-5 space-y-1">
              <li v-for="(error, index) in form.errors.flatten()" :key="index" v-html="error" />
            </ul>
          </div>
        </slot>
      </div>

      <div v-if="dismissible" class="ml-auto pl-3">
        <DismissButton @dismiss="dismiss" />
      </div>
    </div>
  </div>
</template>

<script>
import Base from './../AlertErrors'
import DismissButton from './DismissButton.vue'

export default {
  components: {
    DismissButton
  },

  extends: Base
}
</script>
