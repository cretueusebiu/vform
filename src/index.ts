import Form from './Form'
import Errors from './Errors'

export {
  Form,
  Errors,
  Form as default
}
