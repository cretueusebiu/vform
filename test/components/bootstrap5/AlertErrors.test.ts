import { mount } from '@vue/test-utils'
import { Form } from './../../../src'
import AlertErrors from './../../../src/components/bootstrap5/AlertErrors.vue'

describe('AlertErrors', () => {
  test('render alert with a list of errors', () => {
    const form = new Form({ username: '', password: '' })
    form.errors.set({
      username: ['Username is required'],
      password: ['Password must be at least 6 characters', 'Password does not match']
    })

    const wrapper = mount(AlertErrors, {
      propsData: { form, message: 'Custom message' }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('dismiss alert', () => {
    const form = new Form({ username: '', password: '' })
    form.errors.set({
      username: ['Username is required'],
      password: ['Password must be at least 6 characters', 'Password does not match']
    })

    const wrapper = mount(AlertErrors, {
      propsData: { form, message: 'Custom message' }
    })

    wrapper.find('.btn-close').trigger('click')

    expect(wrapper.html()).toMatchSnapshot()
  })
})
