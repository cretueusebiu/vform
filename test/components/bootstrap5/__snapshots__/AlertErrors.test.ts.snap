// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertErrors dismiss alert 1`] = `
<div role="alert" class="alert alert-danger alert-dismissible"><button type="button" aria-label="Dismiss" class="btn-close"></button>
  <div class="mb-2">Custom message</div>
  <ul class="mb-0 ps-4">
    <li>Username is required</li>
    <li>Password must be at least 6 characters</li>
    <li>Password does not match</li>
  </ul>
</div>
`;

exports[`AlertErrors render alert with a list of errors 1`] = `
<div role="alert" class="alert alert-danger alert-dismissible"><button type="button" aria-label="Dismiss" class="btn-close"></button>
  <div class="mb-2">Custom message</div>
  <ul class="mb-0 ps-4">
    <li>Username is required</li>
    <li>Password must be at least 6 characters</li>
    <li>Password does not match</li>
  </ul>
</div>
`;
